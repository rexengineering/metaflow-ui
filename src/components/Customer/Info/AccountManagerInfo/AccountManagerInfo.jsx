import React from "react";
import { Avatar, Button, makeStyles, Typography } from "@material-ui/core";
import PropTypes from "prop-types";
import clsx from "clsx";
import InfoSection from "../InfoSection";

const useStyles = makeStyles((theme) => ({
  managerInfoContainer: {
    display: "flex",
    justifyContent: "space-between",
    margin: theme.spacing(2, 0),
  },
  avatar: {
    width: theme.spacing(8),
    height: theme.spacing(8),
  },
  fontColor: {
    color: theme.palette.grey["400"],
  },
  accountManagerName: {
    color: theme.palette.common.white,
  },
  transferCallButton: {
    textTransform: "capitalize",
    borderRadius: theme.spacing(10),
  },
}));

function AccountManagerInfo({
  name,
  roleType,
  phoneNumber,
  onTransferCallClick,
  className,
}) {
  const classes = useStyles();
  return (
    <InfoSection className={clsx(classes.fontColor, className)} title="Team">
      <div className={classes.managerInfoContainer}>
        <Avatar className={classes.avatar}>{name?.charAt(0)}</Avatar>
        <div>
          <Typography className={classes.accountManagerName}>{name}</Typography>
          <Typography variant="h6">{roleType}</Typography>
          <Typography variant="h6">{phoneNumber}</Typography>
        </div>
      </div>
      <Button
        onClick={onTransferCallClick}
        fullWidth
        variant="outlined"
        className={classes.transferCallButton}
      >
        Transfer Call
      </Button>
    </InfoSection>
  );
}

AccountManagerInfo.defaultProps = {
  className: "",
};

AccountManagerInfo.propTypes = {
  name: PropTypes.string.isRequired,
  roleType: PropTypes.string.isRequired,
  phoneNumber: PropTypes.string.isRequired,
  onTransferCallClick: PropTypes.func.isRequired,
  className: PropTypes.string,
};

export default AccountManagerInfo;
