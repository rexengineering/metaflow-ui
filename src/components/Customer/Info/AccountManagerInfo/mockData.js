export default {
  name: "Duo Maxwell",
  role: "Account Manager",
  phoneNumber: "555-555-5555",
  address: "123 Main Street, Los Angeles CA 90094",
  onTransferCallClick: () => {},
};
