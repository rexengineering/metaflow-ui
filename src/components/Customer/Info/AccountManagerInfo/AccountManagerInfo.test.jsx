import { render, screen, fireEvent } from "@testing-library/react";
import React from "react";
import AccountManagerInfo from ".";
import managerData from "./mockData";

describe("<AccountManagerInfo />", () => {
  const rendersAccountManagerInfoComponent = (
    onTransferCallClick = managerData.onTransferCallClick
  ) => {
    return (
      <AccountManagerInfo
        address={managerData.address}
        name={managerData.name}
        phoneNumber={managerData.phoneNumber}
        roleType={managerData.role}
        onTransferCallClick={onTransferCallClick}
      />
    );
  };

  it("should render", () => {
    const { asFragment } = render(rendersAccountManagerInfoComponent());
    const initialRender = asFragment();

    expect(initialRender).not.toBeNull();
  });

  it("should match snapshot", () => {
    const { asFragment } = render(rendersAccountManagerInfoComponent());
    const initialRender = asFragment();

    expect(initialRender).toMatchSnapshot();
  });

  it("should handle onTransferCallClick properly", () => {
    const onTransferCallClick = jest.fn();
    render(rendersAccountManagerInfoComponent(onTransferCallClick));
    const transferCallButton = screen.getByRole("button");
    expect(transferCallButton).toBeInTheDocument();
    fireEvent.click(transferCallButton);
    expect(onTransferCallClick).toBeCalled();
  });
});
