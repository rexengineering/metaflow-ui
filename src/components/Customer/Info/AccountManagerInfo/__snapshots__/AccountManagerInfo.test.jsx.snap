// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AccountManagerInfo /> should match snapshot 1`] = `
<DocumentFragment>
  <section
    class="makeStyles-fontColor-10"
  >
    <div
      class="makeStyles-header-13"
    >
      <h6
        class="MuiTypography-root makeStyles-label-14 MuiTypography-h6"
      >
        Team
      </h6>
    </div>
    <div>
      <div
        class="makeStyles-managerInfoContainer-8"
      >
        <div
          class="MuiAvatar-root MuiAvatar-circle makeStyles-avatar-9 MuiAvatar-colorDefault"
        >
          D
        </div>
        <div>
          <p
            class="MuiTypography-root makeStyles-accountManagerName-11 MuiTypography-body1"
          >
            Duo Maxwell
          </p>
          <h6
            class="MuiTypography-root MuiTypography-h6"
          >
            Account Manager
          </h6>
          <h6
            class="MuiTypography-root MuiTypography-h6"
          >
            555-555-5555
          </h6>
        </div>
      </div>
      <button
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined makeStyles-transferCallButton-12 MuiButton-fullWidth"
        tabindex="0"
        type="button"
      >
        <span
          class="MuiButton-label"
        >
          Transfer Call
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </button>
    </div>
  </section>
</DocumentFragment>
`;
